build_config/lib/src/build_config.dart

```diff
@@ -145,24 +145,27 @@ Map<String, T> _normalizeBuilderDefinitions<T>(
         normalizeBuilderKeyDefinition(key, packageName), definition));
 
 String _prettyPrintCheckedFromJsonException(CheckedFromJsonException err) {
-  var yamlMap = err.map as YamlMap;
-
-  var yamlKey = yamlMap.nodes.keys.singleWhere(
-      (k) => (k as YamlScalar).value == err.key,
-      orElse: () => null) as YamlScalar;
-
   var message = 'Could not create `${err.className}`.';
-  if (yamlKey == null) {
-    assert(err.key == null);
-    message = '${yamlMap.span.message(message)} ${err.innerError}';
+  var map = err.map;
+  if (err.key == null) {
+    if (map is YamlMap) {
+      message = '${map.span.message(message)} ${err.innerError}';
+    } else {
+      message = '${err.innerError}';
+    }
   } else {
     message = '$message\nUnsupported value for `${err.key}`: ';
     if (err.message != null) {
       message += '${err.message}\n';
     } else {
       message += '${err.innerError}\n';
     }
-    message = yamlKey.span.message(message);
+    if (map is YamlMap) {
+      var yamlKey = map.nodes.keys.singleWhere(
+          (k) => (k as YamlScalar).value == err.key,
+          orElse: () => null) as YamlScalar;
```

Comment:
nit: if adding an `orElse` which returns `null`, then we should handle the `null` case.